pkg/solver/matcher/matcher.go

```diff
@@ -18,31 +23,54 @@ func (a ListOfResourceOffers) Less(i, j int) bool {
 func (a ListOfResourceOffers) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
 
 func GetMatchingDeals(
+	ctx context.Context,
 	db store.SolverStore,
 	updateJobOfferState func(string, string, uint8) (*data.JobOfferContainer, error),
+	tracer trace.Tracer,
 ) ([]data.Deal, error) {
+	ctx, span := tracer.Start(ctx, "get_matching_deals")
+	defer span.End()
+
 	deals := []data.Deal{}
 
+	// Get resource offers
+	span.AddEvent("db.get_resource_offers.start")
 	resourceOffers, err := db.GetResourceOffers(store.GetResourceOffersQuery{
 		NotMatched: true,
 	})
 	if err != nil {
+		span.SetStatus(codes.Error, "get resource offers failed")
+		span.RecordError(err)
 		return nil, err
 	}
+	span.SetAttributes(attribute.KeyValue{
+		Key:   "resource_offers",
+		Value: attribute.StringSliceValue(data.GetResourceOfferContainerIDs(resourceOffers)),
+	})
```

Comment:
In general, span attributes should be set when the span is created because they are generally considered static information. But we're setting them dynamically here as a part of the initialization leading up to the real work.

Setting these values as span attributes makes it easier to query spans with them.